File: week2_UI-elements/week2_soln/lab2_python/data_types.py
import sys
import cv2

def display(image_object,window_name):
	if(image_object == None):
		print "No image data"
		return
	
	cv2.imshow(window_name,image_object)
	cv2.waitKey(0)
	cv2.destroyAllWindows()

def draw(image_object):
	cv2.line(image_object,(10,10),(520,260),(0,0,255),2)
	cv2.rectangle(image_object,(50,50),(300,150),(0,255,0),2)
	image_object[10][10]=(255,0,0)
	cv2.circle(image_object,(50,70),0,(255,0,0),0)
	cv2.putText(image_object,"CENG 391",(400,400),0,1,(255,0,255),3)
	
	
if __name__ == "__main__":
	image_name = str(sys.argv[1])
	img = cv2.imread(image_name,1)
	draw(img)
	display(img,"Modified Image")
	cv2.imwrite("modified_img.pgm",img)

File: week2_UI-elements/week2_soln/lab2_C++/DataTypes/DataTypes.cpp

#include "opencv2/imgproc/imgproc.hpp"
#include "opencv2/highgui/highgui.hpp"
#include <iostream>

using namespace std;
using namespace cv;


void display(Mat in_img, string win_name ){

  namedWindow(win_name, WINDOW_AUTOSIZE );
  imshow(win_name, in_img);
}

void draw(Mat in_img){
  
  Point pt1(10,10);
  Point pt2(520,260);
    
  line(in_img, pt1, pt2, Scalar(0,0,255), 2);
  rectangle(in_img, Point(50,50), Point(300,150),Scalar(0,255,0),2);
  in_img.at<float>(10,10) = (255,0,0);
  circle(in_img, Point(50,70),0 ,Scalar(255,0,0),0);
  putText(in_img, "CENG 391", Point(400,400),0,1,Scalar(255,0,255),3);
  display(in_img,"modified_img");
  
}

int main( int argc, char** argv )
{
  Mat src = imread("img1.ppm",1);

  if (!src.data) 
    { 
      cout << "Error loading the image" << endl;
      return -1; 
    }
     
  draw(src);
  imwrite("modified_img.ppm",src);
  waitKey(0);

  return 0;
}
 
